game/script-farewells.rpy

```diff
@@ -19,12 +19,30 @@ init python in farewells:
             affinity=store.jn_globals.current_affinity_state,
             additional_properties=[
                 ("is_time_sensitive", store.utils.get_current_session_length().total_seconds() / 60 < 30),
-                ("has_stay_option", not store.jn_globals.player_already_stayed_on_farewell)
+                ("has_stay_option", not store.jn_globals.player_already_stayed_on_farewell and store.jn_globals.current_affinity_state >= 6)
             ]
         )
 
-        # Finally return a random farewell from the remaining pool
-        return random.choice(farewell_pool).label
+        # If pool isn't empty
+        if farewell_pool != []:
```

Comment:
For consistency in code style we can simply do the following
```suggestion
        if farewell_pool:
```

since an empty list will `bool()` down to `False`